### PR TITLE
fix(spells): start resubscribed spells scheduling after full node initialization [fixes NET-459]

### PR DESCRIPTION
**Pull request by kmd-fl**

### CODE REVIEW

File: crates/spell-event-bus/src/bus.rs [hunk 1]
```diff
@@ -208,6 +210,33 @@ impl SpellEventBus {
         let mut sources_channel = futures::stream::select_all(sources);
 
         let mut state = SubscribersState::new();
+        loop {
+            let result: Result<(), BusInternalError> = try {
+                select! {
+                    Some(command) = self.recv_cmd_channel.recv() => {
```
Comment (gurinderu) [inline crates/spell-event-bus/src/bus.rs:216]: Probably it should be inside one loop and one select

Comment (folex) [reply crates/spell-event-bus/src/bus.rs:216]: whoah, yeah, we need to refactor that

Comment (folex) [reply crates/spell-event-bus/src/bus.rs:216]: I think something like `FusedStream` might help

Comment (folex) [reply crates/spell-event-bus/src/bus.rs:216]: `select!` supports conditions in branches, ie

```
<pattern> = <async expression> (, if <precondition>)? => <handler>,
```

So basically you can have a flag that is set in Run and avoid polling `timer` and `sources` if it is not set.

**Review by folex — COMMENTED** [review]
Awesome! This is a very important PR that will make nox more stable.

Just a few things to change before the merge.

File: particle-node/src/node.rs [hunk 2]
```diff
@@ -489,6 +493,11 @@ impl<RT: AquaRuntime> Node<RT> {
             .await
             .wrap_err("builtins deploy failed")?;
 
+        let result = self.spell_event_bus_api.run_scheduling().await;
```
[the remark this answers is not in the document]
Comment (kmd-fl) [reply particle-node/src/node.rs]: Fixed, but it does happen only once, there's no stop.

File: crates/spell-event-bus/src/bus.rs (hunk 1 above)
Comment (kmd-fl) [reply crates/spell-event-bus/src/bus.rs]: Changed on using select conditions. Checked that spells aren't run until `start_scheduling`

File: particle-node/src/node.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (folex) [reply particle-node/src/node.rs]: I mean that `run_scheduling` sounds like something you should call every time you want scheduling to happen.

Like `loop { run_scheduling }`, while `start_scheduling` sounds like a start of a self-sustaining process

**Review by folex — APPROVED** [review]

**Pull request merged**